ddtrace/contrib/internal/azure_functions/patch.py

```diff
@@ -0,0 +1,85 @@
+import azure.functions as azure_functions
+from wrapt import wrap_function_wrapper as _w
+
+from ddtrace import config
+from ddtrace.contrib.trace_utils import int_service
+from ddtrace.contrib.trace_utils import unwrap as _u
+from ddtrace.ext import SpanTypes
+from ddtrace.internal import core
+from ddtrace.internal.schema import schematize_cloud_faas_operation
+from ddtrace.internal.schema import schematize_service_name
+from ddtrace.pin import Pin
+
+
+config._add(
+    "azure_functions",
+    {
+        "_default_service": schematize_service_name("azure_functions"),
+    },
+)
+
+
+def get_version():
+    # type: () -> str
+    return getattr(azure_functions, "__version__", "")
+
+
+def patch():
+    """
+    Patch `azure.functions` module for tracing
+    """
+    # Check to see if we have patched azure.functions yet or not
+    if getattr(azure_functions, "_datadog_patch", False):
+        return
+    azure_functions._datadog_patch = True
+
+    Pin().onto(azure_functions.FunctionApp)
+    _w("azure.functions", "FunctionApp.route", _patched_route)
+
+
+def _patched_route(wrapped, instance, args, kwargs):
+    trigger = "Http"
+
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    def _wrapper(func):
+        function_name = func.__name__
+
+        def wrap_function(req: azure_functions.HttpRequest, context: azure_functions.Context):
+            operation_name = schematize_cloud_faas_operation(
+                "azure.functions.invoke", cloud_provider="azure", cloud_service="functions"
+            )
+            with core.context_with_data(
+                "azure.functions.patched_route_request",
+                span_name=operation_name,
+                pin=pin,
+                service=int_service(pin, config.azure_functions),
```

Comment:
```suggestion
                service=int_service(pin, config.azure_functions),
```
Should azure functions be considered an external service? Generally internal services are services that are fully contained within the application, such as an HTTP server. While external is generally for clients that represent connections to a service outside the application, such as Redis or MySQL.

Reply:
I think internal service makes sense here. The Azure Functions package is essentially a web server intended to run in the Azure Functions environment.

Reply:
okay, makes sense 👍